### PR TITLE
Preserve spacing around text nodes and braces

### LINKED ISSUE #83: Whitespace in HTML macro
How do I create I the equivalent of `Hello Big World` with the html macro?

When I do `html! { <div> Hello Big World </div> }` then the whitespace around the text is stripped. Can it somehow be escaped without using the `text!` macro and variable replacement?

### PR DESCRIPTION
**Pull request by chinedufn**
Fixes #83 

Also refactors a good chunk of the `html-macro` to make it easier to read and a good bit more DRY

### CODE REVIEW

File: crates/html-macro/src/tag.rs [hunk 1]
```diff
@@ -168,31 +221,46 @@ fn parse_text_node(input: &mut ParseStream) -> Result<Tag> {
...
-            }
+        if idx == 0 {
+            start_span = Some(tt.span());
+            most_recent_span = Some(tt.span());
+        }
 
-            text += &tt.to_string();
+        // Insert necessary space in between the tokens in this text node that we're building.
+        // In the real browser DOM there's no difference between one space and many,
+        // so we just insert one.
```
Comment (dbrgn) [inline crates/html-macro/src/tag.rs]: That's not necessarily true! If you use `white-space: pre` or `white-space: pre-wrap` CSS rules then white-space will not be collapsed.

https://developer.mozilla.org/en-US/docs/Web/CSS/white-space

Not sure if that could be a problem or not.

Comment (chinedufn) [reply crates/html-macro/src/tag.rs]: Ahhhh thanks for catching this!

Luckily the fix here is straightforwards enough.

We'll just look at the spans for the text tokens and see how far apart they are - then insert that much space in between then.

And if their spans are on different lines we'll just insert a new line character followed by the amount of spacing that.

---

Also in `text.rs` we'll need to insert the right amount of space and newlines before/after the entire string of text.. vs. right now we just insert a single space

https://github.com/chinedufn/percy/blob/edd4122ccee157566e5118e4c3e74d210eb3a091/crates/html-macro/src/parser/text.rs#L34-L36

---

Thanks for catching this - I'll add some more test cases and get this fixed. Then we should be in awesome shape here!

Comment (chinedufn) [reply crates/html-macro/src/tag.rs]: Took a look and think there's a better solution.. Going to merge this and open a separate issue.

Comment (chinedufn) [reply crates/html-macro/src/tag.rs]: #98

**Pull request merged**